### ISSUE TITLE
Doesn't build on Arch Linux

### ISSUE THREAD

**Issue by kpcyrd** [mun-lang/lld-rs#1]
hi! I'm having trouble building this crate, I have lld 14.0.6.

```
   Compiling lld_rs v130.0.0
The following warnings were emitted during compilation:

warning: wrapper/lld-c.cpp: In function ‘LldInvokeResult mun_lld_link(LldFlavor, int, const char* const*)’:
warning: wrapper/lld-c.cpp:59:45: error: invalid initialization of non-const reference of type ‘llvm::raw_ostream&’ from an rvalue of type ‘bool’
warning:    59 |       result.success = lld::elf::link(args, false, outputStream, errorStream);
warning:       |                                             ^~~~~
warning: In file included from wrapper/lld-c.cpp:2:
warning: /usr/include/lld/Common/Driver.h:42:65: note: in passing argument 2 of ‘bool lld::elf::link(llvm::ArrayRef<const char*>, llvm::raw_ostream&, llvm::raw_ostream&, bool, bool)’
warning:    42 | bool link(llvm::ArrayRef<const char *> args, llvm::raw_ostream &stdoutOS,
warning:       |                                              ~~~~~~~~~~~~~~~~~~~^~~~~~~~
warning: wrapper/lld-c.cpp:65:46: error: invalid initialization of non-const reference of type ‘llvm::raw_ostream&’ from an rvalue of type ‘bool’
warning:    65 |       result.success = lld::wasm::link(args, false, outputStream, errorStream);
warning:       |                                              ^~~~~
warning: /usr/include/lld/Common/Driver.h:52:65: note: in passing argument 2 of ‘bool lld::wasm::link(llvm::ArrayRef<const char*>, llvm::raw_ostream&, llvm::raw_ostream&, bool, bool)’
warning:    52 | bool link(llvm::ArrayRef<const char *> args, llvm::raw_ostream &stdoutOS,
warning:       |                                              ~~~~~~~~~~~~~~~~~~~^~~~~~~~
warning: wrapper/lld-c.cpp:71:29: error: ‘lld::mach_o’ has not been declared
warning:    71 |       result.success = lld::mach_o::link(args, false, outputStream, errorStream);
warning:       |                             ^~~~~~
warning: wrapper/lld-c.cpp:78:46: error: invalid initialization of non-const reference of type ‘llvm::raw_ostream&’ from an rvalue of type ‘bool’
warning:    78 |       result.success = lld::coff::link(args, false, outputStream, errorStream);
warning:       |                                              ^~~~~
warning: /usr/include/lld/Common/Driver.h:32:65: note: in passing argument 2 of ‘bool lld::coff::link(llvm::ArrayRef<const char*>, llvm::raw_ostream&, llvm::raw_ostream&, bool, bool)’
warning:    32 | bool link(llvm::ArrayRef<const char *> args, llvm::raw_ostream &stdoutOS,
warning:       |                                              ~~~~~~~~~~~~~~~~~~~^~~~~~~~

error: failed to run custom build command for `lld_rs v130.0.0`

Caused by:
  process didn't exit successfully: `/home/.../target/release/build/lld_rs-c2d9473ee64f3d31/build-script-build` (exit status: 1)
  --- stdout
  TARGET = Some("x86_64-unknown-linux-gnu")
  OPT_LEVEL = Some("3")
  HOST = Some("x86_64-unknown-linux-gnu")
  cargo:rerun-if-env-changed=CXX_x86_64-unknown-linux-gnu
  CXX_x86_64-unknown-linux-gnu = None
  cargo:rerun-if-env-changed=CXX_x86_64_unknown_linux_gnu
  CXX_x86_64_unknown_linux_gnu = None
  cargo:rerun-if-env-changed=HOST_CXX
  HOST_CXX = None
  cargo:rerun-if-env-changed=CXX
  CXX = None
  cargo:rerun-if-env-changed=CXXFLAGS_x86_64-unknown-linux-gnu
  CXXFLAGS_x86_64-unknown-linux-gnu = None
  cargo:rerun-if-env-changed=CXXFLAGS_x86_64_unknown_linux_gnu
  CXXFLAGS_x86_64_unknown_linux_gnu = None
  cargo:rerun-if-env-changed=HOST_CXXFLAGS
  HOST_CXXFLAGS = None
  cargo:rerun-if-env-changed=CXXFLAGS
  CXXFLAGS = Some("-I/usr/include -std=c++14   -fno-exceptions -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS ")
  cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
  CRATE_CC_NO_DEFAULTS = None
  DEBUG = Some("false")
  CARGO_CFG_TARGET_FEATURE = Some("fxsr,sse,sse2")
  running: "c++" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "-I/usr/include" "-std=c++14" "-fno-exceptions" "-D_GNU_SOURCE" "-D__STDC_CONSTANT_MACROS" "-D__STDC_FORMAT_MACROS" "-D__STDC_LIMIT_MACROS" "-o" "/home/.../target/release/build/lld_rs-8a2d78b5e8aee8ff/out/wrapper/lld-c.o" "-c" "wrapper/lld-c.cpp"
  cargo:warning=wrapper/lld-c.cpp: In function ‘LldInvokeResult mun_lld_link(LldFlavor, int, const char* const*)’:
  cargo:warning=wrapper/lld-c.cpp:59:45: error: invalid initialization of non-const reference of type ‘llvm::raw_ostream&’ from an rvalue of type ‘bool’
  cargo:warning=   59 |       result.success = lld::elf::link(args, false, outputStream, errorStream);
  cargo:warning=      |                                             ^~~~~
  cargo:warning=In file included from wrapper/lld-c.cpp:2:
  cargo:warning=/usr/include/lld/Common/Driver.h:42:65: note: in passing argument 2 of ‘bool lld::elf::link(llvm::ArrayRef<const char*>, llvm::raw_ostream&, llvm::raw_ostream&, bool, bool)’
  cargo:warning=   42 | bool link(llvm::ArrayRef<const char *> args, llvm::raw_ostream &stdoutOS,
  cargo:warning=      |                                              ~~~~~~~~~~~~~~~~~~~^~~~~~~~
  cargo:warning=wrapper/lld-c.cpp:65:46: error: invalid initialization of non-const reference of type ‘llvm::raw_ostream&’ from an rvalue of type ‘bool’
  cargo:warning=   65 |       result.success = lld::wasm::link(args, false, outputStream, errorStream);
  cargo:warning=      |                                              ^~~~~
  cargo:warning=/usr/include/lld/Common/Driver.h:52:65: note: in passing argument 2 of ‘bool lld::wasm::link(llvm::ArrayRef<const char*>, llvm::raw_ostream&, llvm::raw_ostream&, bool, bool)’
  cargo:warning=   52 | bool link(llvm::ArrayRef<const char *> args, llvm::raw_ostream &stdoutOS,
  cargo:warning=      |                                              ~~~~~~~~~~~~~~~~~~~^~~~~~~~
  cargo:warning=wrapper/lld-c.cpp:71:29: error: ‘lld::mach_o’ has not been declared
  cargo:warning=   71 |       result.success = lld::mach_o::link(args, false, outputStream, errorStream);
  cargo:warning=      |                             ^~~~~~
  cargo:warning=wrapper/lld-c.cpp:78:46: error: invalid initialization of non-const reference of type ‘llvm::raw_ostream&’ from an rvalue of type ‘bool’
  cargo:warning=   78 |       result.success = lld::coff::link(args, false, outputStream, errorStream);
  cargo:warning=      |                                              ^~~~~
  cargo:warning=/usr/include/lld/Common/Driver.h:32:65: note: in passing argument 2 of ‘bool lld::coff::link(llvm::ArrayRef<const char*>, llvm::raw_ostream&, llvm::raw_ostream&, bool, bool)’
  cargo:warning=   32 | bool link(llvm::ArrayRef<const char *> args, llvm::raw_ostream &stdoutOS,
  cargo:warning=      |                                              ~~~~~~~~~~~~~~~~~~~^~~~~~~~
  exit status: 1

  --- stderr


  error occurred: Command "c++" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "-I/usr/include" "-std=c++14" "-fno-exceptions" "-D_GNU_SOURCE" "-D__STDC_CONSTANT_MACROS" "-D__STDC_FORMAT_MACROS" "-D__STDC_LIMIT_MACROS" "-o" "/home/.../target/release/build/lld_rs-8a2d78b5e8aee8ff/out/wrapper/lld-c.o" "-c" "wrapper/lld-c.cpp" with args "c++" did not execute successfully (status code exit status: 1).


warning: build failed, waiting for other jobs to finish...
```

**Comment by baszalmstra** [comment]
This crate targets a specific version of lld. The version you are using (130.x) targers lld 13. 

Im in the process of adding support for lld 14.

**Comment by baszalmstra** [comment]
I've just [released version `140.0.0`](https://crates.io/crates/lld_rs/140.0.0) which targets LLD 14. I'm closing this issue. If you continue to have compiler errors, please check your LLD version and reopen an issue.
